### ISSUE TITLE
[Bug]: No media in movie library listed 

### ISSUE THREAD

**Issue by Forceu** [streamyfin/streamyfin#336]
### What happened?

The library overview only says "2 items" for my movie library, the TV show library has the correct amount. When clicking on the library, it says "no items found".

Movies can be found through the home menu or search. This happens on both Android and iOS 

### Reproduction steps

Click on library menu

### Which device and operating system are you using?

Pixel 7a, Android 15
iPad, iOS 18
Jellyfin 10.10.3

### Version

0.23.0

**Comment by fredrikburmester** [comment]
This is due to empty folders or incorrect media in that library folder. Could you check if that's the case? Also compare to jellyfin web. thanks @Forceu

**Comment by Forceu** [comment]
On jellyfin web 728 items are displayed once I click on the library. So far I haven't had any problems with any other apps, is there a way to provide debug logs?

**Comment by nk404030** [comment]
I have a similar Issue. Out of 90 library items on Web, only 16 are displayed in the Streamyfin app. It is a movie library and all the Items missing are the ones that are one layer deeper in the file system than the others. In my case: 

/media/irl/movies/ are shown
/mnt/external-ssd/_jelly/ are shown
/mnt/external-ssd/_jelly/toReencode/ are **not** shown
/mnt/external-ssd/_jelly/toUpload/ are **not** shown

This behavior is consistent on android and ios apps on multiple devices and libraries. Latest Jellyfin 10.10.3, App 0.22.0.
Movies can be found via search.

**Comment by Forceu** [comment]
That would make sense - I have two subfolders labeled HD and SD and they include the movies in further subfolders. That's probably why the app says in the overview that there are two items

**Comment by fredrikburmester** [comment]
Thank you for this, i'll fix that.

**Comment by fredrikburmester** [comment]
Do you guys have special settings to show items nested in subfolders? When i tried creating a library with nested items they don't even show up on the web. 

Example:
/media/ZRMM/Test/2012 (2009)/2012.mkv - **shown**
/media/ZRMM/Test/HD/2067 (2020)/2067.mkv - **not shown**

**Comment by Simon-Eklundh** [comment]
I've tested the following:

I've created a folder for movies. this folder I've linked as "movies" in jellyfin
the way it's formatted is:
movies
-primary
--movie1
--movie2
-sequels
--movie3
--movie4


in streamyfin in the library overview, we see "2 items" in movies, but if I open up the movies overview I see all 4


is there some special formatting you're using @Forceu  and @nk404030

**Comment by Forceu** [comment]
No, I did not change anything in that regard. I just had a look at my settings, do you have `Libraries / Display / Display a folder view to show plain media folders` enabled? for me it is disabled by default I think.

**Comment by Simon-Eklundh** [comment]
@Forceu  I do not have that setting enabled either (I have all default settings rn as I created a second instance just to test your issue)

**Comment by Simon-Eklundh** [comment]
could you maybe describe how your movies are added similar to how I did?

eg:
movies
-primary
--movie1
--movie2
-sequels
--movie3
--movie4

**Comment by Forceu** [comment]
For me my data structure is `[Library]`/`[HD or SD]`/`[Movie name and year]`/`[Movie name and year as file]`

e.g.

`[Library]/HD/12 Angry Men (1957)/12 Angry Men (1957).mkv`

**Comment by Simon-Eklundh** [comment]
so I did that setup but with different names;
movies/primary/moviefolder/movie.mkv

when I open it in streamyfin the library list says "2" in movies, but if I open movies I see all 4I've got

I even tried having one in movies/primary/something/moviefolder/movie.mkv

and still saw all 4

**Comment by nk404030** [comment]
I have the movies on two different drives. They are mounted in the Jellyfin Docker as read-only. `- /mnt/external-ssd/:/mnt/external-ssd:ro` and `- /mnt/jellyfin/:/media:ro `.

media
-irl
--movies
---movie1.mkv                 shown
---movie2.mkv                 shown

external-ssd
-misc
--movie3                          shown
-jellyfin
--folder1
---movie4                        not shown
---movie5                        not shown
--folder2
---movie6                        not shown
---movie7                        not shown

**Comment by Forceu** [comment]
@Simon-Eklundh I would be happy to provide debug logs or network capture, if that helps

**Comment by Simon-Eklundh** [comment]
@Forceu tried both [Library]/[HD or SD]/[Movie name and year]/[Movie name and year as file]
and moving library to HD and SD.

[Library]/[HD or SD]/[Movie name and year]/[Movie name and year as file] works except for us saying there are 2 movies (for me at least)
moving to HD and SD as separate folders 
![image](https://github.com/user-attachments/assets/28b732da-536e-4f41-89f0-df9644b1de6d)
broke jellyfin web... as well as us
(no items in despite the size being 4)

**Comment by Simon-Eklundh** [comment]
@nk404030 how do the folder import look in the streamytest library?

**Comment by nk404030** [comment]
> @nk404030 how do the folder import look in the streamytest library?

its just this, everything else default
![image](https://github.com/user-attachments/assets/b0fecb41-ff7a-4518-93a7-4657e74cbc65)

**Comment by Simon-Eklundh** [comment]
I have a theory @nk404030  and @Forceu  are you on the latest beta?

I was testing on the latest beta and everything worked, then I swapped to the latest version in google play store, and it broke

**Comment by fredrikburmester** [comment]
> I set up a test library and made some screenshots. https://imgur.com/a/2fkyERa

According to this image subfolders are not shown in the web interface either? Am i seeing that correct?

**Comment by nk404030** [comment]
> > I set up a test library and made some screenshots. https://imgur.com/a/2fkyERa
> 
> According to this image subfolders are not shown in the web interface either? Am i seeing that correct?
Yes, i think it just looks two subfolders deep in general.

**Comment by nk404030** [comment]
> I have a theory @nk404030 and @Forceu are you on the latest beta?
> 
> I was testing on the latest beta and everything worked, then I swapped to the latest version in google play store, and it broke

I am on release 0.22.0. I am enrolling in Beta right now to try.

**Comment by Simon-Eklundh** [comment]
oh, and @nk404030  I recommend
movies/moviename/moviename.mkv
for everything

(it's how jellyfin at least used to recommend it)

**Comment by fredrikburmester** [comment]
so folder1 and folder2 are in the temp folder? @nk404030

**Comment by Forceu** [comment]
> I have a theory @nk404030 and @Forceu are you on the latest beta?

No I only downloaded the app from PlayStore and AppStore

**Comment by nk404030** [comment]
> so folder1 and folder2 are in the temp folder? @nk404030

yes

**Comment by fredrikburmester** [comment]
Would anyone be willing to provide a test account? email credentials to fredrik.burmester@gmail.com

**Comment by Simon-Eklundh** [comment]
in that case: I believe it's fixed in the beta (I can't replicate on my beta phone but I can on my none-beta build)

**Comment by nk404030** [comment]
> Would anyone be willing to provide a test account? email credentials to [fredrik.burmester@gmail.com](mailto:fredrik.burmester@gmail.com)

sent you creds

**Comment by nk404030** [comment]
I have the same behavior in Beta mode on google play after reinstalling the app, though I cant seem to verify the Version from inside the app.

**Comment by Forceu** [comment]
Disregard my email, I was able to sign up to the beta as well through the PlayStore and I also have the same behaviour

**Comment by Simon-Eklundh** [comment]
@nk404030  should be visible in the app information in android settings (I think we've discussed adding the version into the app but not gotten that far yet XD)

I have no idea if the playstore beta even works XD

I also forgot the paywall was live (there is a way of live running the app using a pc, buuut I assume Fredrik will come with an answer after testing)

**Comment by Forceu** [comment]
It still says 0.22.0 in the system app info - is that correct for the beta?

**Comment by Simon-Eklundh** [comment]
beta should be 0.23.0

**Comment by Forceu** [comment]
Okay, so Google installed the latest stable release instead of beta, even though it says beta on the play store...

**Comment by Simon-Eklundh** [comment]
yeah, I don't know if we have the playstore beta setup, we mostly use apks for android testing

**Comment by Forceu** [comment]
I compiled the APK from source and with the current commit it is working
